### DISCUSSION TITLE
How to Use Enable/Use Paredit/Pareto?

### DISCUSSION THREAD

**Discussion by veqqq** [lem-project/lem#1599] [Q&A]
At this point, I'm just copying random incantation from people's `init.lisp` files e.g.

```
(require :lem-paredit)
(lem-paredit:paredit-mode t)
(defun pared-hook ()
  (lem-paredit-mode:paredit-mode t))

(define-command lisp-paredit-mode () ()
 (lem-lisp-mode:lisp-mode))
```

but no paredit key bindings work (or exist?)

I also installed pareto then added:

```
(push "~/projects/lisp/lem-pareto/" asdf:*central-registry*)
(asdf:load-system :lem-pareto)
;; Enable Paredit and Pareto along with Lisp mode
(add-hook *find-file-hook*
          (lambda (buffer)
            (when (eq (buffer-major-mode buffer)
                      'lem-lisp-mode:lisp-mode)
              (change-buffer-mode buffer 'lem-paredit-mode:paredit-mode t)
              (change-buffer-mode buffer 'lem-pareto-mode:pareto-mode t))))
```

but the key bindings on pareto don't work. (Actually, they just give single letters like k, but various attempts to chord them like C-k, M-k etc. do nothing.)

For the love of god, I can't figure out how to do basic things with lem..

**Comment by vindarel** [comment]
Hello, for pareto you have to load it then call M-x pareto-mode.

With your snippet maybe paredit and pareto conflict somewhat, like pareto taking precedence? I'd try one then the other.

I just tested pareto, while some keys work (j k C-k maybe more) some didn't (r R c).

It is a contributed module so unfortunately we can't be sure of its quality.

**paredit**

You can also call paredit functions with M-x such as paredit-backward… this works. I can't comment much more I don't know the keybindings :p
